### PR TITLE
Unify the post tags and post categories block under a unique post terms block.

### LINKED ISSUE #31422: Rename Post Tags block to PostNonHierarchicalTerms block and make a Post Tags variation instead.
Right now we have two post taxonomies blocks:

 - postHierarchicalTerms
 - postTags

The former is generic and support post categories variations, the latter is specific to tags. We should consider renaming the latter and make a variation for tags to match the behavior of the hierarchical terms block.

I think this is important to do before shipping these blocks to Core. 

cc @carolinan @mtias

### PR DESCRIPTION
**Pull request by youknowriad**
closes #31422 

This PR does a few things:

 - Renames the existing Post Hierarchical Terms block into just Post terms block.
 - Removes the Post Tags block.
 - Adds a new Post Tags variation for the Post terms block.
 - marks the Post terms block as stable to ship in in 5.8.

These blocks are still experimental, they might be used by some experimental FSE blocks already but I think such a change is needed at the moment.

What do you think?

### CODE REVIEW

File: packages/block-library/src/post-terms/variations.js [hunk 1]
```diff
@@ -14,6 +14,15 @@ const variations = [
 		attributes: { term: 'category' },
 		isActive: ( blockAttributes ) => blockAttributes.term === 'category',
 	},
+
+	{
+		name: 'post_tag',
+		title: __( 'Post Tags' ),
+		description: __( "Display a post's tags." ),
+		icon,
```
Comment (ntsekouras) [inline packages/block-library/src/post-terms/variations.js:22]: We need to set `tag` icon here.

**Review by ntsekouras — COMMENTED** [review]
>These blocks are still experimental, they might be used by some experimental FSE blocks already but I think such a change is needed at the moment.

IMO it's good timing to tidy things up before making this block stable.

Code wise besides the small comments this looks good 👍 

A thing that was before in `PostHierarchicalTerms` block is the usage of `BlockVariationPicker` which was not actually used as we had set `isDefault` to a variation. Same goes here and I don't see any reason not to remove this. Any thoughts?

**Comment by youknowriad** [comment]
> A thing that was before in PostHierarchicalTerms block is the usage of BlockVariationPicker which was not actually used as we had set isDefault to a variation. Same goes here and I don't see any reason not to remove this. Any thoughts?

I think the reason we have this is that we don't want to show the `Post Term` block directly in the inserter, it's too low level.

File: packages/block-library/src/post-terms/variations.js (hunk 1 above)
Comment (youknowriad) [reply packages/block-library/src/post-terms/variations.js:22]: We have both a `category` and `post-categories` icons, we probably need the same for tags which we don't have right now. Maybe a good design follow-up. cc @jasmussen

Comment (jasmussen) [reply packages/block-library/src/post-terms/variations.js:22]: I've set a reminder to follow up here.

**Comment by ntsekouras** [comment]
>I think the reason we have this is that we don't want to show the Post Term block directly in the inserter, it's too low level.

I know and it makes sense, what I'm suggesting is that we could remove `BlockVariationPicker` as is never used right now.

**Comment by youknowriad** [comment]
Oh got it @ntsekouras it should be done now.

File: packages/block-library/src/post-terms/edit.js [hunk 2]
```diff
@@ -92,18 +71,7 @@ export default function PostTermsEdit( {
...
-					icon={ blockType?.icon?.src }
-					label={ blockType?.title }
-					onSelect={ ( variation = defaultVariation ) => {
-						setAttributes( variation.attributes );
-					} }
-					variations={ variations }
-				/>
-			</div>
-		);
+		return __( 'Post Terms block: no term specified.' );
```
Comment (ntsekouras) [inline packages/block-library/src/post-terms/edit.js]: You should wrap this in `<div { ...blockProps }>`.

**Review by ntsekouras — APPROVED** [review]
Looks good 💯 , thank Riad!

**Pull request merged**